qdev_wrappers/tests/dataset/test_doNd.py

```diff
@@ -0,0 +1,165 @@
+"""
+These are the basic black box tests for the doNd functions.
+"""
+
+from qdev_wrappers.dataset.doNd import do0d, do1d, do2d
+from typing import Tuple, List, Optional
+from qcodes.instrument.parameter import Parameter
+from qcodes import config, new_experiment
+from qcodes.utils import validators
+
+import pytest
+
+config.user.mainfolder = "output"  # set ouput folder for doNd's
+new_experiment("doNd-tests", sample_name="no sample")
+
+
+@pytest.fixture()
+def _parameters():
+
+    _param = Parameter('simple_parameter',
+                   set_cmd=None,
+                   get_cmd=lambda: 1)
+
+    _paramComplex = Parameter('simple_complex_parameter',
+                   set_cmd=None,
+                   get_cmd=lambda: 1 + 1j,
+                   vals=validators.ComplexNumbers())
+
+    _param_set = Parameter('simple_setter_parameter',
+                       set_cmd=None,
+                       get_cmd=None)
+
+    return _param, _paramComplex, _param_set
+
+
+def _param_func(_p):
+    """
+    A private utility function.
+    """
+    _new_param = Parameter('modified_parameter',
+                           set_cmd= None,
+                           get_cmd= lambda: _p.get()*2)
+    return _new_param
+
+
+@pytest.fixture()
+def _param_callable(_parameters):
+    _param, _, _ = _parameters
+    return _param_func(_param)
+
+
+def test_param_callable(_param_callable):
+    _param_modified = _param_callable
+    assert _param_modified.get() == 2
+
+
+@pytest.mark.parametrize('period, plot', [(None, True), (None, False),
+                         (1, True), (1, False)])
+def test_do0d(_parameters, _param_callable, period, plot):
```

Comment:
This could be split up into individual tests, for the different scenarios and it could be fused with the other tests by adding more asserts:
so e.g. test do0d with a simple parameter and assert if the output is correct.
Another test could be calling a do0d with a callable and a parameter and check if they are handled in the right order.

Reply:
I would prefer the keep the tests for run-time and return-values and -types seperately. Thus, as I learnt from our offline disscussions, because parametrize creates these seperate tests internally, I would keep the current implementation.

Reply:
Ok, fair enough, that addresses the point of fusing some tests, but how about splitting and asserting order?